src/interface/meshblock_data.hpp

```diff
@@ -291,33 +291,40 @@ class MeshBlockData {
   VariableFluxPack<T> PackVariablesAndFluxes(const std::vector<MetadataFlag> &flags,
                                              vpack_types::StringPair &key);
   VariablePack<T> PackVariables(const std::vector<std::string> &names,
-                                const std::vector<int> &sparse_ids, PackIndexMap &vmap);
+                                const std::vector<int> &sparse_ids, PackIndexMap &vmap,
+                                bool coarse = false);
   VariablePack<T> PackVariables(const std::vector<std::string> &names,
-                                const std::vector<int> &sparse_ids);
+                                const std::vector<int> &sparse_ids, bool coarse = false);
+  VariablePack<T> PackVariables(const std::vector<std::string> &names, PackIndexMap &vmap,
+                                bool coarse = false);
   VariablePack<T> PackVariables(const std::vector<std::string> &names,
-                                PackIndexMap &vmap);
-  VariablePack<T> PackVariables(const std::vector<std::string> &names);
+                                bool coarse = false);
   VariablePack<T> PackVariables(const std::vector<MetadataFlag> &flags,
-                                PackIndexMap &vmap);
-  VariablePack<T> PackVariables(const std::vector<MetadataFlag> &flags);
-  VariablePack<T> PackVariables(PackIndexMap &vmap);
+                                PackIndexMap &vmap, bool coarse = false);
+  VariablePack<T> PackVariables(const std::vector<MetadataFlag> &flags,
+                                bool coarse = false);
+  VariablePack<T> PackVariables(PackIndexMap &vmap, bool coarse = false);
+  // no coarse flag because you need to be able to specify non one-copy variables
```

Comment:
you could probably solve this with some gross template metaprogramming, but let's not...

Reply:
@nmsriram and I actually looked at this for a while. There are other solutions, but this is by far the simplest one.